### PR TITLE
[BUG] Tapping on "Update synced reading lists" alerts the user that reading lists will be synced when they're logged out

### PR DESCRIPTION
**Pull request by nambatee**
https://phabricator.wikimedia.org/T192649

The `else` case on line `231` in `StorageAndSyncingSettingsViewController` is handled in #2256.

### CODE REVIEW

File: Wikipedia/Code/Panels.swift [hunk 1]
```diff
@@ -193,6 +193,11 @@ extension UIViewController {
         }
         let panel = SyncEnabledPanelViewController(showCloseButton: true, primaryButtonTapHandler: primaryButtonTapHandler, secondaryButtonTapHandler: nil, dismissHandler: nil, theme: theme)
         let presenter = self.presentedViewController ?? self
+        if presenter is WMFThemeableNavigationController {
```
Comment (montehurd) [inline Wikipedia/Code/Panels.swift]: Would it be possible to move this `if` to the place where the problematic call to `wmf_showSyncEnabledPanelOncePerLogin` is happening so it doesn't even dip into this func if we don't want it to? (No worries if not.)

File: Wikipedia/Code/Panels.swift [hunk 2]
```diff
@@ -193,6 +193,11 @@ extension UIViewController {
...
+            guard presenter.presentedViewController == nil else {
+                return
```
Comment (montehurd) [inline Wikipedia/Code/Panels.swift]: Do you think an assert makes sense here or is it something we expect to happen?

Comment (nambatee) [reply Wikipedia/Code/Panels.swift]: We expect it to happen. "Sync enabled" panel will be presented when we get a response from the server that sync is enabled for the account. We can be presenting "Your reading lists will be synced in the background" alert at that time.

Comment (montehurd) [reply Wikipedia/Code/Panels.swift]: If you expect it to happen it's fine.

**Pull request merged**